pulsar-client/src/test/java/org/apache/pulsar/client/impl/BatchMessageContainerImplTest.java

```diff
@@ -36,6 +38,33 @@
 
 public class BatchMessageContainerImplTest {
 
+    @Test
+    public void testUpdateMaxBatchSize() {
+        int SHRINK_COOLING_OFF_PERIOD = 10;
+        BatchMessageContainerImpl messageContainer = new BatchMessageContainerImpl();
+        // test expand
+        messageContainer.updateMaxBatchSize(2048);
+        assertEquals(messageContainer.getMaxBatchSize(), 2048);
+
+        // test cooling-off period
+        messageContainer.updateMaxBatchSize(2);
+        assertEquals(messageContainer.getMaxBatchSize(), 2048);
+
+        // test shrink
+        for (int i = 0; i < 15; ++i) {
+            messageContainer.updateMaxBatchSize(2);
+            if (i < SHRINK_COOLING_OFF_PERIOD) {
```

Comment:
Add a case to verify like `messageContainer.updateMaxBatchSize(2000)`;

Reply:
Update~, PTAL again @Jason918